### PR TITLE
PnP SDK part2: Implementation

### PR DESCRIPTION
**Pull request by jspaith**
Bring in the .c files for the PnP SDK, matching the headers already brought in.

This is a refactored version of what's currently in `feature/iot_pnp`.  Largest change is moving pnp layer to just be iot.  There's some additional changes, including leveraging method parsing for PnP commands and of course matching up to other header changes from original.

For process of this work:

Part1 of PR brought in the header files.
Part2 **is this current PR**.  It brings in the .c files implementing the code.
Part3 will bring in the tests.
Part4 will bring in the samples.  At that point we'll bring everything over to `feature/iot_pnp`.

Note that the code that's being merged has been tested and successfully run against the samples already [here](https://github.com/jspaith/azure-sdk-for-c/tree/pnp_refactor_poc/).  The split up is done to make this more manageable to review.

### CODE REVIEW

**Review by ericwol-msft — APPROVED** [review]
thanks for adding comments

File: sdk/src/azure/iot/CMakeLists.txt [hunk 1]
```diff
@@ -26,6 +26,14 @@ target_link_libraries(az_iot_common
 
 add_library (az::iot::common ALIAS az_iot_common)
 
+# Azure IoT Plug and Play source files for use when implementing a 
+# PnP device, i.e. one that implements a model using Digital Twins 
+# Definition Language (DTDL).  These files can be removed needed.
```
Comment (ericwol-msft) [inline sdk/src/azure/iot/CMakeLists.txt:31]: removed needed?

File: sdk/src/azure/iot/az_iot_hub_client_properties.c [hunk 2]
```diff
@@ -0,0 +1,482 @@
...
+      }
+    }
+    else
+    {
+      return AZ_OK;
+    }
+  }
+}
+
+// verify_valid_json_position makes sure that our the az_json_reader
```
Comment (danewalton) [inline sdk/src/azure/iot/az_iot_hub_client_properties.c:326]:
```suggestion
// verify_valid_json_position makes sure that the az_json_reader
```

File: sdk/src/azure/iot/CMakeLists.txt (hunk 1 above)
Comment (jspaith) [reply sdk/src/azure/iot/CMakeLists.txt:31]: Yup - removing whole comment block per dwalton feedback.

File: sdk/src/azure/iot/az_iot_hub_client_methods.c [hunk 3]
```diff
@@ -88,7 +88,7 @@ AZ_NODISCARD az_result az_iot_hub_client_methods_response_get_publish_topic(
   _az_PRECONDITION_VALID_SPAN(client->_internal.iot_hub_hostname, 1, false);
   _az_PRECONDITION_VALID_SPAN(request_id, 1, false);
   _az_PRECONDITION_NOT_NULL(mqtt_topic);
-  _az_PRECONDITION(mqtt_topic_size > 0);
+  _az_PRECONDITION(mqtt_topic_size);
```
[the remark this answers is not in the document]
Comment (jspaith) [reply sdk/src/azure/iot/az_iot_hub_client_methods.c:91]: Yeah, I agree too since it's not pretending to be a ptr computation.  fixing.

**Review by hihigupt — APPROVED** [review]
:shipit:

**Pull request merged**